### DISCUSSION TITLE
Render a template conditionally

### DISCUSSION THREAD

**Discussion by TheoStefou** [ckan/ckan#7516] [Q&A]
How would I take control of a template and add my custom logic to it? What I mean is the following:

Let's say I want to add some extra logic to the resource template before it is rendered. If my check passes, I want the default resource template to be rendered, otherwise my custom "error template" is rendered.
The problem is that I cannot just add a blueprint that "overrides" the path to resources, because then I would not be able to pass the correct parameters to the default resource template if my check passes. Redirecting to the resource path would just cause an infinite amount of redirects because my own blueprint would have control over the resource url.

The flow would be something like this:
```
def my_resource_render(context, data_dict):
  if custom_check_user_allowed_to_see_resource(context, data_dict):
    return toolkit.render('/path/to/resource_template.html', ?which vars go here?)
  else:
    return toolkit.render('/path/to/my/error_template.html', {extra_vars})

```

**Comment by Pooya-Oladazimi** [comment]
The better solution is to find the template file and the block that contains your target code and then load your content there conditionally. 
You can define template helper functions in your plugin to check the condition while rendering:
https://docs.ckan.org/en/2.9/theming/templates.html?highlight=template%20helper#template-helper-functions

**Comment by pdelboca** [comment]
Yeah, overriding full views in CKAN is still not a perfect experience, so doing what you want to do requires some work.

Some thoughts:
 1. Handle access at a resource level is a common requirement (I assume this is what you are trying to do). There are several people doing it so I would ask around how they are solving this issue. 
 2. It is perfectly okay to just override the view in your extension by Copy-Paste the whole view from core and then adding custom logic to it. This is of course not ideal, but Blueprints and views in general takes care of the UX so it is okay to just re do it to match your project's expectations.
 3. @Pooya-Oladazimi suggestion is also valid. You can play around with a helper like `h.check_resource_access(...)` and then coditionaly render what's needed.
